Add evil-tex-ts

Prgebish opened the pull request:
Tree-sitter based LaTeX text objects and evil-surround integration for Evil mode.

**Repository:** https://github.com/Prgebish/evil-tex-ts

## Summary

evil-tex-ts provides LaTeX-specific text objects for Emacs Evil mode using tree-sitter for accurate parsing (Emacs 29.1+). It is a modern reimplementation of [evil-tex](https://github.com/iyefrat/evil-tex) without AUCTeX dependency.

**Features:**
- Text objects: environments (`ie`/`ae`), commands (`ic`/`ac`), math (`im`/`am`), delimiters (`id`/`ad`), sections, sub/superscripts
- Toggles: environment asterisk, math mode cycling, delimiter sizing, command asterisk
- evil-surround integration with keymaps for environments, delimiters, and accents

**Dependencies:** `evil`, `treesit` (built-in since Emacs 29.1)

## Checklist

- [x] Package builds successfully (`make recipes/evil-tex-ts`)
- [x] `package-lint` passes (only `with-eval-after-load` warnings for optional integrations)
- [x] GPL-3.0 license
- [x] Documentation in README.org

riscy commented:
Thanks for this.  Could you complete (& verify) the original checklist?:

- [] The package is released under a [GPL-Compatible Free Software License](https://www.gnu.org/licenses/license-list.en.html#GPLCompatibleLicenses)
- [] I've read [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)
- [] I've used the latest version of [package-lint](https://github.com/purcell/package-lint) to check for packaging issues, and addressed its feedback
- [] My elisp byte-compiles cleanly
- [] I've used `M-x checkdoc` to check the package's documentation strings
- [] I've built and installed the package using the instructions in [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)

riscy commented:
Thanks a ton for accommodating.  Here's a quick first pass on my side -

⸺ `evil-tex-ts.el` with package-lint 20251205.1720:
```
5 issues found:
2029:1: warning: `with-eval-after-load' is for use in configurations, and should rarely be used in packages.
2048:1: warning: `with-eval-after-load' is for use in configurations, and should rarely be used in packages.
2186:1: warning: `with-eval-after-load' is for use in configurations, and should rarely be used in packages.
2217:1: warning: `with-eval-after-load' is for use in configurations, and should rarely be used in packages.
2221:1: warning: `with-eval-after-load' is for use in configurations, and should rarely be used in packages.
```
I think the `with-eval-after-load` is actually unnecessary here (your package explicitly requires `evil` up front).  I'd rewrite the keybindings as:
```elisp
(defvar evil-tex-ts-inner-text-objects-map 
  (let ((keymap (make-sparse-keymap)))
    (define-key keymap "e" #'evil-tex-ts-inner-environment)
    (define-key keymap "c" #'evil-tex-ts-inner-command)
    (define-key keymap "m" #'evil-tex-ts-inner-math)
    (define-key keymap "d" #'evil-tex-ts-inner-delimiter)
    (define-key keymap "^" #'evil-tex-ts-inner-superscript)
    (define-key keymap "_" #'evil-tex-ts-inner-subscript)
    (define-key keymap "S" #'evil-tex-ts-inner-section)
    keymap)
    "Inner text object keymap for `evil-tex-ts'.")

(defvar evil-tex-ts-outer-text-objects-map
  (let ((keymap (make-sparse-keymap)))
    (define-key keymap "e" #'evil-tex-ts-outer-environment)
    (define-key keymap "c" #'evil-tex-ts-outer-command)
    (define-key keymap "m" #'evil-tex-ts-outer-math)
    (define-key keymap "d" #'evil-tex-ts-outer-delimiter)
    (define-key keymap "^" #'evil-tex-ts-outer-superscript)
    (define-key keymap "_" #'evil-tex-ts-outer-subscript)
    (define-key keymap "S" #'evil-tex-ts-outer-section)
    keymap)
  "Outer text object keymap for `evil-tex-ts'.")
```

⸺ `evil-tex-ts.el` with [melpazoid](https://github.com/riscy/melpazoid):
```
- evil-tex-ts.el#L234: Consider `beginning-of-line`
- evil-tex-ts.el#L1804: Top-level `setq` should usually be replaced by `defvar` or `defconst`
- evil-tex-ts.el#L1905: Top-level `setq` should usually be replaced by `defvar` or `defconst`
- evil-tex-ts.el#L1954: Top-level `setq` should usually be replaced by `defvar` or `defconst`
```
In this case, these top-level `setq` nags look like they can be solved using a similar `defvar keymap` pattern as I wrote above.

Prgebish commented:
Hi,

  Thank you for the thorough review and the helpful suggestions! I've
addressed all the feedback:

  package-lint issues (with-eval-after-load):
  - Removed all with-eval-after-load 'evil wrappers since evil is an
explicit dependency
  - Removed with-eval-after-load 'which-key for the replacement-alist
configuration
  - Removed the late evil-surround integration hook (users should ensure
evil-surround is loaded before activating the mode)

  melpazoid issues:
  - Replaced (goto-char (line-beginning-position)) with beginning-of-line
  - Replaced all top-level setq with defvar using the let-initialization
pattern you suggested for the keymaps

  All 204 tests pass and package-lint reports no warnings now.

  Thanks again for the guidance!

пн, 29 дек. 2025 г. в 02:09, Chris Rayner ***@***.***>:

> *riscy* left a comment (melpa/melpa#9751)
> <https://github.com/melpa/melpa/pull/9751#issuecomment-3694970520>
>
> Thanks a ton for accommodating. Here's a quick first pass on my side -
>
> ⸺ evil-tex-ts.el with package-lint 20251205.1720:
>
> 5 issues found:
> 2029:1: warning: `with-eval-after-load' is for use in configurations, and should rarely be used in packages.
> 2048:1: warning: `with-eval-after-load' is for use in configurations, and should rarely be used in packages.
> 2186:1: warning: `with-eval-after-load' is for use in configurations, and should rarely be used in packages.
> 2217:1: warning: `with-eval-after-load' is for use in configurations, and should rarely be used in packages.
> 2221:1: warning: `with-eval-after-load' is for use in configurations, and should rarely be used in packages.
>
> I think the with-eval-after-load is actually unnecessary here (your
> package explicitly requires evil up front). I'd rewrite the keybindings
> as:
>
> (defvar evil-tex-ts-inner-text-objects-map
>   (let ((keymap (make-sparse-keymap)))
>     (define-key keymap "e" #'evil-tex-ts-inner-environment)
>     (define-key keymap "c" #'evil-tex-ts-inner-command)
>     (define-key keymap "m" #'evil-tex-ts-inner-math)
>     (define-key keymap "d" #'evil-tex-ts-inner-delimiter)
>     (define-key keymap "^" #'evil-tex-ts-inner-superscript)
>     (define-key keymap "_" #'evil-tex-ts-inner-subscript)
>     (define-key keymap "S" #'evil-tex-ts-inner-section)
>     keymap)
>     "Inner text object keymap for `evil-tex-ts'.")
>
> (defvar evil-tex-ts-outer-text-objects-map
>   (let ((keymap (make-sparse-keymap)))
>     (define-key keymap "e" #'evil-tex-ts-outer-environment)
>     (define-key keymap "c" #'evil-tex-ts-outer-command)
>     (define-key keymap "m" #'evil-tex-ts-outer-math)
>     (define-key keymap "d" #'evil-tex-ts-outer-delimiter)
>     (define-key keymap "^" #'evil-tex-ts-outer-superscript)
>     (define-key keymap "_" #'evil-tex-ts-outer-subscript)
>     (define-key keymap "S" #'evil-tex-ts-outer-section)
>     keymap)
>   "Outer text object keymap for `evil-tex-ts'.")
>
> ⸺ evil-tex-ts.el with melpazoid <https://github.com/riscy/melpazoid>:
>
> - evil-tex-ts.el#L234: Consider `beginning-of-line`
> - evil-tex-ts.el#L1804: Top-level `setq` should usually be replaced by `defvar` or `defconst`
> - evil-tex-ts.el#L1905: Top-level `setq` should usually be replaced by `defvar` or `defconst`
> - evil-tex-ts.el#L1954: Top-level `setq` should usually be replaced by `defvar` or `defconst`
>
> In this case, these top-level setq nags look like they can be solved
> using a similar defvar keymap pattern as I wrote above.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/melpa/melpa/pull/9751#issuecomment-3694970520>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AKPCU3S6CFTCZWRU4CXDUO34EATH3AVCNFSM6AAAAACQCSYBOKVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZTMOJUHE3TANJSGA>
> .
> You are receiving this because you are subscribed to this thread.Message
> ID: ***@***.***>
>

riscy commented:
I'll merge this now - welcome to MELPA. :)
